pallets/amm/src/lib.rs

```diff
@@ -328,19 +333,21 @@ pub mod pallet {
 
 			let (remove_amount_a, remove_amount_b) = liquidity_out;
 
-			ensure!(
-				T::Currency::free_balance(asset_a, &pair_account) >= remove_amount_a,
-				Error::<T>::InsufficientPoolAssetBalance
-			);
-			ensure!(
-				T::Currency::free_balance(asset_b, &pair_account) >= remove_amount_b,
-				Error::<T>::InsufficientPoolAssetBalance
-			);
-
 			let liquidity_left = total_shares
 				.checked_sub(liquidity_amount)
 				.ok_or(Error::<T>::InvalidLiquidityAmount)?;
 
+			let user_liquidity_left = user_shares
+				.checked_sub(liquidity_amount)
+				.ok_or(Error::<T>::InvalidLiquidityAmount)?;
+
+			if total_shares != liquidity_amount {
```

Comment:
this condition is probably redundant ?  can total shares be less than min pool limit ?

Reply:
If it was not there you wouldn't be able to withdraw everything since the ensure inside makes sure that you can't leave amount smaller than min liquidity limit left in the pool. We should probably add documentation here stating why we are checking it against users liquidity (that's because users could interlock each other if they left smaller amount than min liquidity limit)

Reply:
It could also be `liquidity_left == zero` which is basically the same thing.